PowerShell/README.md

```diff
@@ -0,0 +1,371 @@
+
+```powershell
+# Instead of this long line:
+Get-ChildItem -Path (Join-Path -Path $PSScriptRoot -ChildPath 'Arc' -Resolve) -Recurse -Filter '*.ps1' -Exclude $exclude -Include $include
+
+# Break it into two:
+$path = Join-Path -Path $PSScriptRoot -ChildPath 'Arc' -Resolve
+Get-ChildItem -Path $path -Recurse -Filter '*.ps1' -Exclude $exclude -Include $include
+```
+
+If a line that calls another command is longer than 100 characters, put each parameter after the first on a new line, indented to line up with the first parameter. When breaking a command across multiple lines, don't put more than one parameter on a line.
+
+Instead of this:
+
+```
+Invoke-Robocopy -Source $binSourcePath `
+                -Destination $destinationBinPath `
+                -IncludeFiles $binWhitelist `
+                -ExcludeFiles $ExcludeBinFiles `
+                -Retry 12 `
+                -RetryWaitSeconds 5 `
+                -Mirror
+```
+
+When a pipeline is longer than 100 characters, put each command/step of the pipeline on its own line:
+
+```powershell
+Get-ChildItem -Recurse | 
+    Sort-Object -Property 'Size' |
+    Where-Object { $_.Size -gt 1mb } |
+    Select-Object -ExpandProperty 'FullName'
+```
+
+When assigning the result of a pipeline to a variable, and the pipeline is longer than 100 characters, break each command onto its own line, and put the first command of the pipeline on the line, indented one level:
+
+```powershell
+$largeFiles = 
+    Get-ChildItem -Recurse | 
+    Sort-Object -Property 'Size' |
+    Where-Object { $_.Size -gt 1mb } |
+    Select-Object -ExpandProperty 'FullName'
+```
+
+When using a script block to assign the value of a variable, indent the contents of the script block one level below the indentation of the variable:
+
+```powershell
+$searchPaths = & {
+    Join-Path -Path (Get-Location).ProviderPath -ChildPath $powerShellModulesDirectoryName
+    Join-Path -Path $PSScriptRoot -ChildPath '..\Modules' -Resolve
+}
+```
+
+Variable names should begin with a lowercase letter.
+
+All blocks (except script blocks) that require curly braces must have the opening and closing curly braces on their own lines:
+
+```powershell
+if( $true )
+{
+    # Something
+}
+else
+{
+    # Something else
+}
+```
+
+
+When negating a statement, use the -not  operator.
+
+```powershell
+# Instead of using !
+if( !$false ) 
+{
+}
+
+# Use -not
+if( -not $false )
+{
+}
+```
+
+# Output
+
+Never use `Write-Host`. It can't be redirected, captured, or ignored.
+
+Never use `Write-Output` to return logging information. You should only ever return objects.
+
+Use `Write-Information` for progess-type logging information the user should always see.
+
+Use `Write-Verbose` for messages the user should see if they're trying to track down a problem or understand more about what your code is doing.
+
+Use `Write-Debug` for messages developer-only messages.
+
+## Script Template

```

Comment:
This markdown isn't rendering correctly, shows up as just `## Script Template` in the rendered view.

Reply:
There was some weird character between `##` and `Script`. It's now a space.